### ISSUE TITLE
Invalid thread access error when opening kettle files in a multi-user environment

### ISSUE THREAD

**Issue by HiromuHota** [HiromuHota/pentaho-kettle#30]
How to reproduce

1. User A opens webSpoon
2. User B opens webSpoon
3. User A tries to open a kettle file on a local filesystem

NPE happens when User A tries to open a kettle file on a repository.

```
org.eclipse.swt.SWTException: Invalid thread access
	at org.eclipse.swt.SWT.error(SWT.java:3720)
	at org.eclipse.swt.SWT.error(SWT.java:3643)
	at org.eclipse.swt.SWT.error(SWT.java:3614)
	at org.eclipse.swt.widgets.Widget.error(Widget.java:1018)
	at org.eclipse.swt.widgets.Widget.checkWidget(Widget.java:956)
	at org.eclipse.swt.widgets.Control.getShell(Control.java:258)
	at org.eclipse.swt.widgets.Control.checkBackground(Control.java:624)
	at org.eclipse.swt.widgets.Control.createWidget(Control.java:213)
	at org.eclipse.swt.widgets.Control.<init>(Control.java:206)
	at org.eclipse.swt.widgets.Scrollable.<init>(Scrollable.java:74)
	at org.eclipse.swt.widgets.Composite.<init>(Composite.java:89)
	at org.pentaho.di.ui.spoon.AbstractGraph.<init>(AbstractGraph.java:61)
	at org.pentaho.di.ui.spoon.trans.TransGraph.<init>(TransGraph.java:404)
	at org.pentaho.di.ui.spoon.delegates.SpoonTransformationDelegate.addTransGraph(SpoonTransformationDelegate.java:180)
	at org.pentaho.di.ui.spoon.Spoon.addTransGraph(Spoon.java:8751)
	at org.pentaho.di.ui.spoon.TransFileListener.open(TransFileListener.java:111)
	at org.pentaho.di.ui.spoon.Spoon.openFile(Spoon.java:4593)
	at org.pentaho.di.ui.spoon.Spoon.lastFileSelect(Spoon.java:1866)
	at org.pentaho.di.ui.spoon.Spoon$6.run(Spoon.java:1786)
	at org.eclipse.jface.action.Action.runWithEvent(Action.java:493)
	at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:575)
	at org.eclipse.jface.action.ActionContributionItem.access$2(ActionContributionItem.java:492)
	at org.eclipse.jface.action.ActionContributionItem$5.handleEvent(ActionContributionItem.java:403)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:109)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:687)
	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:594)
	at org.eclipse.swt.widgets.Display.executeNextEvent(Display.java:1217)
	at org.eclipse.swt.widgets.Display.runPendingMessages(Display.java:1198)
	at org.eclipse.swt.widgets.Display.safeReadAndDispatch(Display.java:1181)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:1173)
	at org.eclipse.rap.rwt.application.AbstractEntryPoint.createUI(AbstractEntryPoint.java:69)
	at org.eclipse.rap.rwt.internal.lifecycle.RWTLifeCycle.createUI(RWTLifeCycle.java:177)
	at org.eclipse.rap.rwt.internal.lifecycle.RWTLifeCycle$UIThreadController.run(RWTLifeCycle.java:290)
	at java.lang.Thread.run(Thread.java:745)
	at org.eclipse.rap.rwt.internal.lifecycle.UIThread.run(UIThread.java:107)
```

**Comment by HiromuHota** [comment]
The root cause of this problem is that the static Spoon.class is globally shared among different sessions. It should be session-unique and not shared.
Related to #23 .

**Comment by HiromuHota** [comment]
Resolved by 893d6dbd4a66c64310cce5fb4af73a976c313982.
